protocol/test-sequencing.md

```diff
@@ -0,0 +1,216 @@
+- Host the same sequencer admin RPC as op-node, to hande sequencer rotation:
+    - `admin_startSequencer`: start
+    - `admin_stopSequencer`: stop
+    - `admin_sequencerActive`: sequencer state
+    - `admin_postUnsafePayload`: publish a block, for failover
+    - `admin_overrideLeader`: used by op-conductor
+    - `admin_conductorEnabled`: used by op-conductor
+- Connect to geth P2P, to read/write the shared tx-pool.
+- Support the DA-throttling RPC.
+  Or better, implement proper multi-block DA congestion safe-guards.
+
+### Engine mode
+
+To not only build blocks locally, we can make the service connect to a regular engine-API endpoint,
+and use that to build the blocks.
+
+This won't support the test capabilities, but does enable migration away from op-node sequencing.
+
+The service essentially becomes an interface for the op-conductor setup to work with any EL builder setup.
+
+**This includes rollup-boost**: without modifications to rollup-boost, we can migrate sequencing away from the op-node.
+[Flashblocks utilizes a sidecar design](https://github.com/ethereum-optimism/design-docs/pull/86),
+called rollup-boost, that enables flash-blocks to be flexibly attached between the services.
+
+### Interop experimentation
+
+With some communication between the build processes, we can coordinate transaction-inclusion across chains.
+E.g.:
+- Host multiple builders in the same Go process.
+- Connect multiple instances with streams.
+
+Following a stream (the "shred" draft in older interop specs),
+we can signal when events happen, and when they are rolled back (callstack revert, or tx not included).
+The other builder can then use the assumed events, or even await them, to build the block of the other chain.
+
+In the extreme, this allows synchronous EVM calls between chains, without interop spec changes.
+
+With the remote-state block-building, and the tx-pool access without running a full node,
+this could potentially also serve as a replica-node, following the "horizontal tx-pool" design.
+
+## Resource Usage
+
+For testing, this separates experimental code from the production sequencing,
+to not affect resource-usage or related processing timing constraints.
+
+By encapsulating the sequencer work, we can monitor and scale it independent of regular verifier work.
+
+Long-term this can open up more scalable block-building for interop.
+
+## Single Point of Failure and Multi Client Considerations
+
+If we decide to productionize this service after test-tooling, we can support the same set of admin RPCs,
+allowing sequencer-rotation to work the same as it does today.
+
+# Alternatives Considered
+
+## Extending op-geth
+
+The block-building described here conflict with that within the `miner` package of geth:
+- the `miner` package block-building recommits blocks.
+- the `miner` package pulls in transactions from the pool eagerly.
+- the `miner` package is a difficult diff to upstream geth, something we do not want to expand.
```

Comment:
+1, extending `op-geth` is not viable for new block building policies if we want it to be extensible and also minimal diff.